src/ItemTable.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import { useGetItemsQuery } from '../app/services/api';
+import { Button } from 'primereact/button';
+import { Dialog } from 'primereact/dialog';
+import { InputText } from 'primereact/inputtext';
+import { MultiSelect } from 'primereact/multiselect';
+import { DataTable, DataTableSortEvent } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+
+interface Item {
+  id: number;
+  n: string;
+  c: string[];
+}
+
+const ItemTable: React.FC = () => {
+  const { data: items = [] } = useGetItemsQuery(undefined); 
+  const [sortField, setSortField] = useState<string | undefined>(undefined); 
+  const [sortOrder, setSortOrder] = useState<1 | -1 | null>(null); 
+  const [editItem, setEditItem] = useState<Item | null>(null); 
+  const [displayDialog, setDisplayDialog] = useState(false); 
+
+  const onSort = (e: DataTableSortEvent) => {
+    setSortField(e.sortField);
+    setSortOrder(e.sortOrder === undefined || e.sortOrder === 0 ? null : e.sortOrder);
+  };
+
+  const onEdit = (item: Item) => {
+    setEditItem(item);
+    setDisplayDialog(true);
+  };
+
+  const hideDialog = () => {
+    setEditItem(null);
+    setDisplayDialog(false);
+  };
+
+  const actionBodyTemplate = (rowData: Item) => (

```

Comment:
We pass both this and some other functions form this file to a component so we might want to wrap each one of them into `useCallback`